internal/discovery/secret_lister.go

```diff
@@ -6,27 +6,47 @@ import (
 	listersv1 "k8s.io/client-go/listers/core/v1"
 )
 
+type SecretListerer interface {
```

Comment:
The `Listerer` name seems slightly strange to me. Seeing that this interface seems very similar to https://pkg.go.dev/k8s.io/client-go/listers/core/v1#SecretLister, perhaps we could borrow the names used upstream

Reply:
I'm not convinced either of the name but because `listerer` is not a valid word and my spell checker in vscode complains :).
I thought about it but in the case of this interface it returns lister , it doesn't list . I would expect that a lister list so could be confusing i think.

Reply:
I guess the authors of go-client faced a similar issue. I think the way they resolved it is by making the top-level `Lister` not having a `List` method, but rather a `Secrets` method that returns the real `Lister`.

Might be confusing but if we cnanot thing about a better name I'd just do what they did :)

Moreover if we name the method the same way, this means their object will also satisfy our interface. Which I'm not sure if it will be useful at any point but who knows :)